graphql-webclient/src/main/java/graphql/kickstart/spring/webclient/boot/GraphQLWebClient.java

```diff
@@ -16,7 +16,7 @@ static GraphQLWebClient newInstance(WebClient webClient, ObjectMapper objectMapp
 
   <T> Mono<T> post(String resource, Map<String, Object> variables, Class<T> returnType);
 
-  <T> Mono<T> post(GraphQLRequest<T> request);
+  Mono<GraphQLResponse> post(GraphQLRequest<?> request);
```

Comment:
You're breaking existing API here and making it work differently compared to the `flux` version too. Because of this change the benefit of having the result be mapped to specific return type is gone too. Perhaps the return type conversion should be moved from here into the `GraphQLResponse` entirely as optional method, something like:
```java
graphqlResponse.get("$.data.simple", Simple.class)
```

Reply:
Thanks for commenting, are you suggesting making all Post/Flux method return a` GraphQLResponse` object, and moving the return type conversion into `GraphQLResponse ` as an optional method? 
My motivation is to have a post/flux method that returns only a GraphQLResponse so that we can apply our own result mapping and error handling logic.

Reply:
Seems like an [exception is thrown ](https://github.com/graphql-java-kickstart/graphql-spring-webclient/blob/master/graphql-webclient/src/main/java/graphql/kickstart/spring/webclient/boot/GraphQLResponse.java#L25)if there is even a single error and any other relevant data that the graphql service returned may be lost because data is not mapped in the exception. I think there may be value in adding a separate method to the interface to return GraphQLResponse.

Reply:
Was thinking about it a bit more and will probably make some more breaking changes anyway.